### PR TITLE
Run clang-format and clang-tidy on eBPF code

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Run clang-format and clang-tidy on eBPF code. It runs automatically during the `system-probe.test` target. Issues do not fail test runs as the moment. This will allow us to gradually fix issues highlighted, rather than forcing all development to stop.

Incorporating these checks into a pre-commit hook is ideal, but is complicated by the need to have path-dependent compiler flags. I'm leaving that to a later PR.

### Motivation

Ensure ebpf C code is following a consistent style and format.

### Additional Notes

We cannot run clang-tidy on the BCC code, since it doesn't compile with `clang`.

### CODE REVIEW

**Comment by albertvaka** [comment]
Don't know why but the tests are failing with:
```
./pkg/ebpf/c/bpf_helpers.h:4:10: fatal error: 'uapi/linux/bpf.h' file not found
#include <uapi/linux/bpf.h>
```

**Comment by mx-psi** [comment]
Just a comment: It's not necessary/desirable to tackle this on this PR but we have pre-commit hooks where we could also add this so that people can easily check this client-side https://github.com/DataDog/datadog-agent/blob/master/.pre-commit-config.yaml

File: pkg/ebpf/c/tracer-ebpf.c [hunk 1]
```diff
@@ -19,7 +19,10 @@
 #define LOAD_CONSTANT(param, var) asm("%0 = " param " ll" \
                                       : "=r"(var))
 
-enum telemetry_counter{tcp_sent_miscounts, missed_tcp_close, udp_send_processed, udp_send_missed};
+enum telemetry_counter { tcp_sent_miscounts,
+    missed_tcp_close,
+    udp_send_processed,
+    udp_send_missed };
```
Comment (sunhay) [inline pkg/ebpf/c/tracer-ebpf.c:25]: I think we we want `tcp_sent_miscounts` on its own line so that its formatted similarly to the enum in `tcp_states.h`

Comment (brycekahle) [reply pkg/ebpf/c/tracer-ebpf.c:25]: I'll see what option I need to set for it to format that the way we want. 😄

**Review by mx-psi — APPROVED** [review]

**Review by lebauce — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Pull request merged**